folsom/src/main/java/com/spotify/folsom/MemcacheClientBuilder.java

```diff
@@ -103,7 +107,7 @@
 
   private int connections = 1;
   private boolean retry = true;
-  private Executor executor = DefaultExecutor.INSTANCE;
```

Comment:
@danielnorberg I think this line would make it non-lazy

Reply:
Indeed, so only change this line? The memoized supplier doesn’t seem like it would’ve avoided this issue.

Reply:
Yes, that would be an option too. I suppose this is more of a stylistic change. `Suppliers.memoize()` could be considered more obvious than the solution with a private static inner class.

Reply:
Yeah, I can buy that.